src/Network/HTTPClient/Factory/HttpClient.php

```diff
@@ -123,7 +123,7 @@ public function createClient(HandlerStack $handlerStack = null): ICanSendHttpReq
 		$resolver->setMaxRedirects(10);
 		$resolver->setRequestTimeout(10);
 		// if the file is too large then exit
-		$resolver->setMaxResponseDataSize(1000000);
+		$resolver->setMaxResponseDataSize($this->config->get('system', 'max_response_data_size', 1000000));
```

Comment:
Please consider avoiding the `system` config category in favor of a more descriptive one like `performance` for example. We've been stuck with `system` as an almost catch-all for most configuration keys but new ones can break free from this obsolete paradigm.

Reply:
Ops, sorry! I for got that one. Will change to e.g. `http_client`?

Reply:
Or `url_resolver`?

Reply:
Above in same code are more keys with `system` as category. But those are old keys and surely one day need moving to that new category, too.

Reply:
Not sure, I find it a little too code-specific. I would like us to create the categories from the administrator standpoint, which would make this fall under the performance section. We should also warn that lowering this limit could partially break some federation features like some large posts not being able to be imported.

Reply:
It seem to have fixed it for @schmacker 's problem. Then next step is to separate requesting (large) posts from (large) files. The later one then can be configured at administrator's discretion. Okay, I can change it to `performance` then.